source/runbooks/dr-process.html.md.erb

```diff
@@ -0,0 +1,362 @@
+
+
+## Internal Escalation
+
+
+**Simon can you help fill in this**
+
+
+Inform the following channels of the situation and put forward a contactable member of staff from our team.
+
+
+`#ask-modernisation-platform`
+
+
+`#modernisation-platform`
+
+
+`#modernisation-platform-alerts`
+
+Following [this]( https://user-guide.modernisation-platform.service.justice.gov.uk/runbooks/manage-an-incident.html#incident-process) process.
+
+## Disaster Recovery Types
+
+
+**Let’s work the problem, people. Let’s not make things worse by guessing.**
+
+
+- London has been lost but **no services need to migrate**. [Click here](#london-has-been-lost-but-returning-soon).
+- London has been lost, and **all services need to migrate**. [Click here](#disaster-recovery-running-outside-of-london).
+- Loss of a **single account**. [Click here](#single-account).
+
+
+## London has been lost but returning soon.
+
+
+### Getting a decision
+
+
+Once a ticket has been raised with AWS, and we have engaged with them and have total confirmation that the infrastructure will be returning within X number of minutes or hours, we will need to pass this information on for an informed decision to be made above.
+
+
+Only when we have this, can we update users on what is next.
+
+
+### Priority list
+
+
+If there is limited availability, which is a possibility, these are the priority accounts that need to be checked before we can focus on member-accounts.
+
+
+
+|Account|Information|
+|----------------- |------------------------------------------------ |
+|Core-Shared-Services|Creates the core shared services account resources. |
+| Core-Network-Services | Creates the core networking account resources; this is the highest priority. |
+|Core-Logging|Creates the core logging account resources |
+|Core-Security|Creates the core security account resources |
+|Cloud-Platform-Transit-Gateways|Across all regions. |
+|Core-VPC-*|Creates the core VPC resources in the VPC accounts, needed after the core-networking-services. |
+| Modernisation-Platform | Creates key resources such as S3 state buckets for the platform |
+
+
+### Test Accounts
+
+
+If there is limited availability, power off all infrastructure in the test accounts. All are useless until the platform is running.
+
+
+
+## Single Account
+
+Things to remember
+
+- Dont forget the statefile, this could cause issues when rebuilding and generating plans.
+
+
+### Will it be returning
+
+
+If this is a test account, waiting may be the easier idea. If AWS can promise us this is a temporary issue with an estimated fix time, we can pass this to the team and agree our next step.
+
+
+### Rebuild account
+
+
+Assuming this is a production account with no time given to return, please follow the steps.
+
+
+#### Workflows
+
+
+Rather than rebuilding an account through workflows, I would run steps manually to easier debug issues, however, technically, you could start to run the account creation process with our github workflows.
```

Comment:
How exactly would you create an account manually? I don't think our roles allow us to create an account, so it would need to be done as part of the account creation workflow I believe.

This documentation should then mention migration of the code of that account to the new account and running the terraform in MP repo first and then doing the same in the env repo.


If the whole account was lost, what do you suggest for backups? Even if the backups were replicate across regions, if a single account is gone, so are the backups.

Reply:
I'm not sure it's possible but could the account be created in another environment first so it kind if sits there unloved and unused until needed. Hopefully it will never be needed.

Reply:
My point was, it would be impossible to create an account manually using our roles as it calls the root account as part of it.

You need the workflows to provision the skeleton account (to create the resources and files in both mp and env repos, it's just part of the account creation) prior running the existing code. This is still bare minimum, as it creates the standard files and infra that are part of account creation. Then for the migration of the old code, you can skip the env repo part, as the bare minimum will be created with the workflow, but don't forget the MP repo part, as this completes the account recovery.

Understanding of the account creation process is easier and less prone to error when done using automation.

Reply:
I'll reword what I have put,